### PR TITLE
Genegraph with redis

### PR DESCRIPTION
**Pull request by theferrit32**
Features added:
- genegraph helm deploy can optionally start a redis deployment exposed on a cluster service with a persistent disk
- genegraph deployment can optionally connect to a redis host for VRS variant cache access
- genegraph can optionally start with a persistent volume with configurable size. Other still-supported options are using container storage (is ephemeral and within host root filesystem) or local ssd.
- the genegraph docker CMD can be overriden by setting `docker_command` in a values file. Useful for main function args.

### CODE REVIEW

File: helm/charts/clingen-genegraph/templates/redis.yaml [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+          persistentVolumeClaim:
+            claimName: {{ .Release.Name }}-redis-pvc
+      containers:
+        - image: {{ .Values.genegraph_redis_image }}
+          imagePullPolicy: Always
+          name: redis
+          resources:
+            requests:
+              memory: 1Gi
+              cpu: 0.5
```
Comment (sjahl) [inline helm/charts/clingen-genegraph/templates/redis.yaml:30]: If we're not sure yet what the limits should be, then it's probably fine to leave them out for now. However, it's probably a good idea to set limits here if we know what they should be set to.

File: helm/charts/clingen-genegraph/templates/redis.yaml [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+      storage: 50Gi
+---
+# https://github.com/kubernetes/examples/blob/master/staging/persistent-volume-provisioning/README.md
+apiVersion: storage.k8s.io/v1
+kind: StorageClass
+metadata:
+  name: {{ .Release.Name }}-redis-storageclass
+  labels:
+    {{- include "genegraph-redis.labels" . | nindent 4 }}
+provisioner: kubernetes.io/gce-pd
```
Comment (sjahl) [inline helm/charts/clingen-genegraph/templates/redis.yaml:74]: One thing I might set on this storage class is `allowVolumeExpansion: true` ; the gce-pd provisioner supports growing volumes online if that's set. In a case where we need more disk space in a pinch, you can increase the `spec.resources.requests.storage` attribute on the PVC and it'll resize the volume. Might be worth configuring the genegraph PV with the same setting if you think that running out of disk space will be a possibility at any point.

https://kubernetes.io/docs/concepts/storage/storage-classes/#allow-volume-expansion

**Review by sjahl — COMMENTED** [review]
Just a couple comments inline, otherwise looks pretty good

File: helm/charts/clingen-genegraph/templates/redis.yaml (hunk 2 above)
Comment (theferrit32) [reply helm/charts/clingen-genegraph/templates/redis.yaml:74]: Thanks, that could be nice to have, I'll add that.

Is it necessary to change the `apiVersion` of this? That doc says it needs `v1.11`

Comment (sjahl) [reply helm/charts/clingen-genegraph/templates/redis.yaml:74]: Nope, the storageclass API should stay V1 -- the v1.11 note in the docs is talking about the kubernetes cluster version.

**Pull request merged**